src/coreclr/debug/dbgutil/machoreader.cpp

```diff
@@ -229,9 +229,8 @@ MachOModule::ReadLoadCommands()
                 m_segments.push_back(segment);
 
                 // Calculate the load bias for the module. This is the value to add to the vmaddr of a
-                // segment to get the actual address. For shared modules, this is 0 since those segments
-                // are absolute address.
-                if (segment->fileoff == 0 && segment->filesize > 0)
+                // segment to get the actual address.
+                if (strcmp(segment->segname, SEG_TEXT) == 0)
```

Comment:
Is there any guarantee that the first segment is the text segment, and that there's only one ever (the spec seems to imply that there is always at least ? Otherwise we may get here more than once or have a wrong offset. 

Also, what happens with the bias of data segments? Granted, this was already never considered if the first segment was always text.